StorageTests/TestSQLiteBookmarks.swift

```diff
@@ -77,6 +79,188 @@ class TestSQLiteBookmarks: XCTestCase {
+            "bkm2", "http://example.org/2", "Bookmark 2", BookmarkRoots.MobileFolderGUID,
+            "bkm3", "http://example.org/3", "Bookmark 3", BookmarkRoots.MobileFolderGUID,
+            "bkm4", "http://example.org/4", "Bookmark 4", BookmarkRoots.MobileFolderGUID,
+        ]
+
+        let structureQuery =
+        "INSERT INTO \(TableBookmarksLocalStructure) (parent, child, idx) VALUES " +
+        "(?, ?, ?), " +
+        "(?, ?, ?), " +
+        "(?, ?, ?), " +
+        "(?, ?, ?) "
+
+        let structureArgs: Args = [
+            BookmarkRoots.MobileFolderGUID, "bkm1", 0,
+            BookmarkRoots.MobileFolderGUID, "bkm2", 1,
+            BookmarkRoots.MobileFolderGUID, "bkm3", 2,
+            BookmarkRoots.MobileFolderGUID, "bkm4", 3,
+        ]
+
+        let bufferQuery =
+        "INSERT INTO \(TableBookmarksBuffer) (guid, type, bmkUri, title, parentid, parentName, server_modified) VALUES " +
+        "(?, \(BookmarkNodeType.folder.rawValue), NULL, ?, ?, '', ?), " +
+        "(?, \(BookmarkNodeType.folder.rawValue), NULL, ?, ?, '', ?), " +
+        "(?, \(BookmarkNodeType.bookmark.rawValue), ?, ?, ?, '', ?) "
+
+        let bkmBufModified = Date.now()
+        let bufferArgs: Args = [
+            BookmarkRoots.RootGUID, "", BookmarkRoots.RootGUID, Date.now(),
+            BookmarkRoots.MobileFolderGUID, "Mobile Bookmarks", BookmarkRoots.RootGUID, Date.now(),
+            "bkmbuf", "http://example.org/5", "Bookmark 5", BookmarkRoots.MobileFolderGUID, bkmBufModified
+        ]
+
+        let bufferStructureQuery =
+        "INSERT INTO \(TableBookmarksBufferStructure) (parent, child, idx) VALUES " +
+        "(?, ?, ?), " +
+        "(?, ?, ?), " +
+        "(?, ?, ?)"
+
+        let bufferStructureArgs: Args = [
+            BookmarkRoots.RootGUID, BookmarkRoots.RootGUID, 0,
+            BookmarkRoots.RootGUID, BookmarkRoots.MobileFolderGUID, 0,
+            BookmarkRoots.MobileFolderGUID, "bkmbuf", 0,
+        ]
+
+        db.run([
+            (sql: localQuery, args: localArgs),
+            (sql: structureQuery, args: structureArgs),
+            (sql: bufferQuery, args: bufferArgs),
+            (sql: bufferStructureQuery, args: bufferStructureArgs),
+            ]).succeeded()
+
+        let mobileRoot = BookmarkMirrorItem.folder(BookmarkRoots.MobileFolderGUID, modified: Date.now(), hasDupe: false, parentID: BookmarkRoots.MobileFolderGUID,
+                                                   parentName: nil, title: "Mobile Bookmarks", description: nil, children: ["bkmbuf", "bkm1", "bkm2", "bkm3"])
+        let op = BufferUpdatedCompletionOp(bufferValuesToMoveFromLocal: Set(["bkm1", "bkm2", "bkm3"]), mobileRoot: mobileRoot, modifiedTime: 123456)
+
+        let mergedBookmarks = MergedSQLiteBookmarks(db: db)
+        let expectation = self.expectation(description: #function)
+        mergedBookmarks.applyBufferUpdatedCompletionOp(op) >>> {
```

Comment:
In tests, you can flatten these out by using `.succeeded()` here, and `.value.successValue!` instead of `>>==`.